### PR TITLE
[RHELC-1174] Update remediation key to match leapp report metadata

### PR DESCRIPTION
**Pull request by r0x0d**
The remediation key was singular in our report, and leapp one is in plural. We are matching the key in our report to remove this workaround from the pre-conversion script that is executed with the rhc-worker-script.

Jira Issues: [RHELC-1174](https://issues.redhat.com/browse/RHELC-1174)

Checklist

- [ ] PR has been tested manually in a VM (either author or reviewer)
- [ ] Jira issue has been made public if possible
- [x] `[RHELC-]` is part of the PR title 
- [ ] GitHub label has been added to help with Release notes 
- [ ] PR title explains the change from the user's point of view
- [ ] Code and tests are documented properly
- [ ] The commits are squashed to as few commits as possible (without losing data) 
- [ ] When merged: Jira issue has been updated to `Release Pending` if relevant

### CODE REVIEW

**Review by pr-watson — APPROVED** [review]
Changes look good to me

**Review by SpyTec — APPROVED** [review]
Needs another rebase due to PRs I merged today

**Comment by r0x0d** [comment]
> Needs another rebase due to PRs I merged today

Will do it today.

**Comment by r0x0d** [comment]
> > Needs another rebase due to PRs I merged today
> 
> Will do it today.

Done the rebase, @SpyTec.

**Review by pr-watson — APPROVED** [review]

**Comment by SpyTec** [comment]
Some leftovers
```
        [2024-01-05T09:13:29-0500] TASK - [Pre-conversion analysis report] ***********************************
        [2024-01-05T09:13:29-0500] DEBUG - /var/run/lock/convert2rhel.pid PID 1609 unlocked.
        Traceback (most recent call last):
          File "/usr/lib/python3.6/site-packages/convert2rhel/main.py", line 129, in main_locked
            disable_colors=logger_module.should_disable_color_output(),
          File "/usr/lib/python3.6/site-packages/convert2rhel/actions/report.py", line 219, in summary
            combined_results_and_message = get_combined_results_and_message(results)
          File "/usr/lib/python3.6/site-packages/convert2rhel/actions/report.py", line 135, in get_combined_results_and_message
            "remediations": action_value["result"]["remediation"],
        KeyError: 'remediation'
        
        During handling of the above exception, another exception occurred:
        
        Traceback (most recent call last):
          File "/usr/lib/python3.6/site-packages/convert2rhel/main.py", line 166, in main_locked
            return _handle_main_exceptions(process_phase, pre_conversion_results)
          File "/usr/lib/python3.6/site-packages/convert2rhel/main.py", line 195, in _handle_main_exceptions
            disable_colors=logger_module.should_disable_color_output(),
          File "/usr/lib/python3.6/site-packages/convert2rhel/actions/report.py", line 219, in summary
            combined_results_and_message = get_combined_results_and_message(results)
          File "/usr/lib/python3.6/site-packages/convert2rhel/actions/report.py", line 135, in get_combined_results_and_message
            "remediations": action_value["result"]["remediation"],
        KeyError: 'remediation'
        
        During handling of the above exception, another exception occurred:
        
        Traceback (most recent call last):
          File "/usr/bin/convert2rhel", line 11, in <module>
            load_entry_point('convert2rhel==1.6.1', 'console_scripts', 'convert2rhel')()
          File "/usr/lib/python3.6/site-packages/convert2rhel/initialize.py", line 82, in run
            return main.main()
          File "/usr/lib/python3.6/site-packages/convert2rhel/main.py", line 87, in main
            return main_locked()
          File "/usr/lib/python3.6/site-packages/convert2rhel/main.py", line 173, in main_locked
            actions.report.summary_as_txt(pre_conversion_results)
          File "/usr/lib/python3.6/site-packages/convert2rhel/actions/report.py", line 304, in summary_as_txt
            combined_results_and_message = get_combined_results_and_message(results)
          File "/usr/lib/python3.6/site-packages/convert2rhel/actions/report.py", line 135, in get_combined_results_and_message
            "remediations": action_value["result"]["remediation"],
        KeyError: 'remediation'
        FAILED
```

**Review by bocekm — DISMISSED** [review]
We need a new schema version. The latest one (https://github.com/oamg/convert2rhel/blob/main/schemas/assessment-schema-1.1.json) has `remediation` a it needs to have `remediations`.

**Comment by r0x0d** [comment]
Will work on this as soon as possible!

File: schemas/assessment-schema-1.2.json [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+                "variables": {
+                    "description": "Information about this particular system that may be used to template the diagnosis and remediation fields.",
+                    "type": "object",
+                    "patternProperties": {
+                        "^[A-Za-z0-9_]+$": {
+                        }
+                    }
+                }
+            },
+            "required": ["title", "description", "diagnosis", "id", "remediation", "variables"]
```
Comment (pr-watson) [inline schemas/assessment-schema-1.2.json:105]:
```suggestion
            "required": ["title", "description", "diagnosis", "id", "remediations", "variables"]
```
I believe this needs to be updated to remediations as well

**Comment by pr-watson** [comment]
@r0x0d changes look good, just made one suggestion

File: schemas/assessment-schema-1.2.json (hunk 1 above)
Comment (r0x0d) [reply schemas/assessment-schema-1.2.json:105]: good catch!

File: schemas/assessment-schema-1.2.json [hunk 2]
```diff
@@ -0,0 +1,150 @@
+{
+    "$schema": "https://json-schema.org/draft/2020-12/schema",
+    "$id": "https://raw.githubusercontent.com/oamg/convert2rhel/main/schemas/assessment-schema-1.1.json",
```
Comment (SpyTec) [inline schemas/assessment-schema-1.2.json:3]:
```suggestion
    "$id": "https://raw.githubusercontent.com/oamg/convert2rhel/main/schemas/assessment-schema-1.2.json",
```

Comment (SpyTec) [reply schemas/assessment-schema-1.2.json:3]: Just this change, otherwise code LGTM

**Review by SpyTec — CHANGES_REQUESTED** [review]
Schema diff LGTM, just one change needed
```➜ diff assessment-schema-1.1.json assessment-schema-1.2.json
92c92
<                 "remediation": {
---
>                 "remediations": {
105c105
<             "required": ["title", "description", "diagnosis", "id", "remediation", "variables"]
---
>             "required": ["title", "description", "diagnosis", "id", "remediations", "variables"]
```

File: schemas/assessment-schema-1.2.json (hunk 2 above)
Comment (r0x0d) [reply schemas/assessment-schema-1.2.json]: Thanks for catching that too!

**Comment by SpyTec** [comment]
Merging once sanity tests are green enough -- won't be for 7 or Alma 8 atm due to infra and Alma changes

**Review by SpyTec — APPROVED** [review]

**Pull request merged**